specutils/spectra/spectral_coordinate.py

```diff
@@ -0,0 +1,433 @@
+                              "{}.".format(auto_dist), AstropyUserWarning)
+
+                frame = SkyCoord(frame, distance=auto_dist).frame
+
+            # If the observer frame does not contain information about the
+            # velocity of the system, assume that the velocity is zero in the
+            # system.
+            if 's' not in frame.data.differentials:
+                warnings.warn(
+                    "No velocity defined on frame, assuming {}.".format(
+                        u.Quantity([0, 0, 0], unit=u.km/u.s)),
+                    AstropyUserWarning)
+
+                frame = SkyCoord(frame, radial_velocity=0 * u.km / u.s).frame
+
+        return frame
+
+    @property
+    def observer(self):
+        """
+        The coordinate frame from which the observation was taken.
+
+        Returns
+        -------
+        `BaseCoordinateFrame`
+            The astropy coordinate frame representing the observation.
+        """
+        return self._observer
+
+    @observer.setter
+    def observer(self, value):
+        value = self._validate_frame(value, True)
+        self._observer = value
+
+        # Reset radial velocity and redshift values
+        self.radial_velocity = None
+
+    @property
+    def target(self):
+        """
+        The coordinate frame of the object being observed.
+
+        Returns
+        -------
+        `BaseCoordinateFrame`
+            The astropy coordinate frame representing the target.
+        """
+        return self._target
+
+    @target.setter
+    def target(self, value):
+        value = self._validate_frame(value, False)
+        self._target = value
+
+        # Reset radial velocity and redshift values
+        self.radial_velocity = None
+
+    @property
+    def rest(self):
+        """
+        The rest value of the spectrum used for transformations to/from
+        velocity space.
+
+        Returns
+        -------
+        `Quantity`
+            Rest value as an astropy `Quantity` object.
+        """
+        return self._rest
+
+    @rest.setter
+    @u.quantity_input(value=['length', 'frequency', 'energy', 'speed', None])
+    def rest(self, value):
+        """
+        New rest value needed for velocity-space conversions.
+
+        Parameters
+        ----------
+        value : `Quantity`
+            Rest value.
+        """
+        self._rest = value
+
+    @property
+    def velocity_convention(self):
+        """
+        The defined convention for conversions to/from velocity space.
+
+        Returns
+        -------
+        str
+            One of 'optical', 'radio', or 'relativistic' representing the
+            equivalency used in the unit conversions.
+        """
+        return self._velocity_convention
+
+    @velocity_convention.setter
+    def velocity_convention(self, value):
+        """
+        New velocity convention used for velocity space conversions.
+
+        Parameters
+        ----------
+        value
```

Comment:
todo: add docs here?  Or is the expectation that this docstring won't be visible?